Make pg_rewind work with encrypted WAL

dAdAbird opened the pull request:
The main goal was to pass pg_rewind tests to find and fix WAL-related issues. Along with that, now we have pg_rewind working with encrypted WAL. There are no changes in the user interface.

This PR omits one of the pg_rewind tests. That fails due to pg_basebackup (see [PG-1603](https://perconadev.atlassian.net/browse/PG-1603)).

For PG-1037, PG-1412

[PG-1603]: https://perconadev.atlassian.net/browse/PG-1603?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

dAdAbird committed:
Re-encrypt WAL segments when copying data to the new timeline

We use TLI to calculate encryption IV, therefore, data in segments has
to be re-encrypted when copied to the new timeline.

For PG-1412

dutow commented on src/bin/pg_rewind/parsexlog.c:
```diff
@@ -356,6 +360,11 @@ SimpleXLogPageRead(XLogReaderState *xlogreader, XLogRecPtr targetPagePtr,
+							targetHistory[private->tliIndex].tli,
+							xlogreadsegno, WalSegSz);
+#else
 	/* Read the requested page */
```
Maybe this comment could go before the ifdef?

dAdAbird replied:
done

dutow reviewed (commented):
For the second commit:

>These tests are a copy of upstream's pg_rewind tests with small changes
in setup_cluster to set WAL encryption on.

Can you split this commit into two, one that just copies the files, and another that contains only the required modifications?

dAdAbird committed:
Add pg_rewind TAP tests

These tests are a carbon copy of upstream's pg_rewind tests. Changes
for WAL encryption will be added in the following commit.
`standby_source` test is not copied as it fails because of changes
needed in pg_basebackup.

dAdAbird committed:
Add WAL encryption to TDE pg_rewind tests

Now the TDE version of pg_rewind tests runs with the WAL encryption ON

dAdAbird committed:
Make pg_rewind work with encrypted WAL
